install_files/ansible-base/files/validate-gpg-key.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+set -e
+set -o pipefail
+
+# Verifies a key at at given path matches a fingerprint
+# Does so by importing the key into gpg2 and checking output
+
+# Exit codes
+# 0 - fingerprint matches key
+# 1 - otherwise
+
+declare -r key_location="$1"
+declare -r fingerprint="$2"
+
+: {key_location:?'A path to a key must be provided (arg 1)'}
+: {fingerprint:?'A fingerprint must be provided (arg 2)'}
+
+# check if fingerprint is formatted correctly
+echo "$fingerprint" | egrep -q '^[0-9A-F]{40}$'
+
+# validate key against fingerprint
+gpg2 --import "$key_location"
+gpg2 --list-keys --fingerprint "$fingerprint" | \
+  grep 'Key fingerprint =' | \
+  sed -e 's/ //g' | \
+  grep -q "$fingerprint"
```

Comment:
Importing the key and then checking the expected fingerprint doesn't test that the fingerprint matches the file. Consider the following:

1. Generate valid keypair and list pubkey filename and correct fingerprint in `prod-specific.yml`.
2. Run playbooks to set config on servers.
3. Generate _new_ keypair and update the pubkey filename, but _not_ the fingerprint in `prod-specific.yml`.
4. Run playbooks to set config on servers.

In step 4, the playbook will complete successfully, because the old fingerprint was never removed from the keyring. So the test passes when it shouldn't. A reasonable solution here would be to set up a temporary GNUPGHOME dir, and import the key there, so it contains _only_ the key currently being imported.